Block/Service/AbstractTagsBlockService.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ClassificationBundle\Block\Service;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\ClassificationBundle\Model\ContextInterface;
+use Sonata\ClassificationBundle\Model\ContextManagerInterface;
+use Sonata\ClassificationBundle\Model\TagInterface;
+use Sonata\ClassificationBundle\Model\TagManagerInterface;
+use Sonata\CoreBundle\Model\Metadata;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Christian Gripp <mail@core23.de>
+ */
+abstract class AbstractTagsBlockService extends AbstractClassificationBlockService
+{
+    /**
+     * @var TagManagerInterface
+     */
+    private $tagManager;
+
+    /**
+     * @var AdminInterface
+     */
+    private $tagAdmin;
+
+    /**
+     * @param string                  $name
+     * @param EngineInterface         $templating
+     * @param ContextManagerInterface $contextManager
+     * @param TagManagerInterface     $tagManager
+     * @param AdminInterface          $tagAdmin
+     */
+    public function __construct($name, EngineInterface $templating, ContextManagerInterface $contextManager, TagManagerInterface $tagManager, AdminInterface $tagAdmin)
```

Comment:
No. http://symfony.com/doc/current/contributing/code/standards.html#structure

> Declare all the arguments on the same line as the method/function name, no matter how many arguments there are;

Plus, there is no hard limit on PSR-2, only soft.